packages/ui/src/components/va-button/VaButton.vue

```diff
@@ -1,241 +1,211 @@
+    // classes
+    const wrapperClassComputed = computed(() => ({ 'va-button__content--loading': props.loading }))
 
-        background = props.gradient ? getGradientBackground(color) : color
-      }
+    const isSlotContentPassed = computed(() => !!slots.default?.()?.[0]?.children)
+    const isOneIcon = computed(() => !!((props.iconRight && !props.icon) || (!props.iconRight && props.icon)))
+    const computedClass = useBem('va-button', () => ({
+      ...pick(props, ['disabled', 'block', 'loading', 'round', 'plain']),
+      small: props.size === 'small',
+      normal: !props.size || props.size === 'medium',
+      large: props.size === 'large',
+      opacity: props.textOpacity < 1,
+      'icon-only': !slots.default && isOneIcon.value,
+      bordered: !!props.borderColor,
+      'no-label': !isSlotContentPassed.value,
+      focused: isFocused.value,
+    }))
 
-      if (focusState.value) {
-        const alpha = props.outline ? -0.8 : -0.7
-        const lightness = 10
-        const color = isTransparentBackground.value ? shiftHSLAColor(colorComputed.value, { a: alpha }) : shiftHSLAColor(colorComputed.value, { l: lightness })
+    // styles
+    const isLowContrastBg = computed(() => props.plain || isLightBackground(colorComputed.value, props.backgroundOpacity))
+    const { textColorComputed } = useTextColor(colorComputed, isLowContrastBg.value)
 
-        background = props.gradient ? getGradientBackground(color) : color
-      }
+    const backgroundComputed = useButtonBackground(colorComputed, isPressed, isHovered)
+    const contentColorComputed = useButtonTextColor(textColorComputed, colorComputed, isPressed, isHovered)
 
-      return {
-        color: textColorComputed.value,
-        borderColor,
-        background,
-      }
-    })
+    const computedStyle = computed(() => ({
+      borderColor: props.borderColor ? getColor(props.borderColor) : 'transparent',
+      ...backgroundComputed.value,
+      ...contentColorComputed.value,
+    }))
 
-    const focus = () => button.value?.focus()
-    const blur = () => button.value?.blur()
+    const publicMethods = { focus, blur }
 
     return {
       button,
       tagComputed,
-      hrefComputed,
       computedClass,
       computedStyle,
-      computedType,
       textColorComputed,
-      loaderSize,
-      focusState,
-      hoverState,
-      focus,
-      blur,
+      loaderSizeComputed,
+      attributesComputed,
+      wrapperClassComputed,
+      iconAttributesComputed,
+
+      ...publicMethods,
     }
   },
 })
 </script>
 
 <style lang='scss'>
-@import '../../styles/resources';
 @import 'variables';
+@import '../../styles/resources';
 
 .va-button {
+  position: relative;
+  margin: var(--va-button-margin);
+  padding: var(--va-button-padding);
   display: var(--va-button-display);
-  align-items: var(--va-button-align-items);
   justify-content: var(--va-button-justify-content);
+  align-items: var(--va-button-align-items);
+  border: var(--va-button-border);
```

Comment:
maybe split border into multiple style params? As you have border color separately and it kinda conflicts with 'border'